be/src/storage/lake/tablet_manager.h

```diff
@@ -67,7 +68,8 @@ class TabletManager {
 
     StatusOr<VersionedTablet> get_tablet(int64_t tablet_id, int64_t version);
 
-    StatusOr<CompactionTaskPtr> compact(int64_t tablet_id, int64_t version, int64_t txn_id);
+    StatusOr<CompactionTaskPtr> compact(int64_t tablet_id, int64_t version, int64_t txn_id,
+                                        CompactionTaskContext& context);
```

Comment:
```suggestion
    StatusOr<CompactionTaskPtr> compact(CompactionTaskContext* context);
```
`CompactionTaskContext` contains `tablet_id`, `version` and `txn_id` already.